ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -153,19 +201,25 @@ LogSegment getNextSegment(LogSegment segment) {
   }
 
   /**
-   * @param name the name of the segment required.
-   * @return a {@link LogSegment} with {@code name} if it exists.
+   * Returns the {@link LogSegment} that is logically before the given {@code segment}.
+   * @param segment the {@link LogSegment} whose "previous" segment is required.
+   * @return the {@link LogSegment} that is logically before the given {@code segment}.
    */
-  LogSegment getSegment(String name) {
-    return segmentsByName.get(name);
+  LogSegment getPrevSegment(LogSegment segment) {
```

Comment:
Will sync up with you offline as to how the abstraction works or who owns the logical ordering of log segments. I see new APIs being added in PersistentIndex like getLogSegmentCount(), getAbsolutePositionInLogForOffset(Offset) which work closely with log segments. So, why not use segmentsByName  or the iterator in Log and expose a method in Log ? I see that going forward index dictates how the log looks from a store's perspective. If yes, can you explain how segmentsByName and the iterator will different from what the index sees?